gogroup.go

```diff
@@ -0,0 +1,18 @@
+package main
+
+// Run a group of goroutines and report the first error.
+func gogroup[T any](args []T, f func(T) error) error {
+	errch := make(chan error)
+
+	gopher := func(arg T) {
+		if err := f(arg); err != nil {
+			errch <- err
+		}
+	}
+
+	for _, arg := range args {
+		go gopher(arg)
+	}
+
+	return <-errch
+}
```

Comment:
I could go either way on this, but this feels like a premature abstraction to me - it could just be written directly into main() without having a separate utility function.

If you do decide to leave it here, can you add a copyright/license comment at the top (following the example in other files) and also add your name to AUTHORS?

Reply:
Unrefactored